lib/matplotlib/text.py

```diff
@@ -632,27 +632,40 @@ def _get_wrapped_text(self):
         # Build the line incrementally, for a more accurate measure of length
         line_width = self._get_wrap_line_width()
         wrapped_str = ""
-        line = ""
-
-        for word in self.get_text().split(' '):
-            # New lines in the user's test need to force a split, so that it's
-            # not using the longest current line width in the line being built
-            sub_words = word.split('\n')
-            for i in range(len(sub_words)):
-                current_width = self._get_rendered_text_width(
-                    line + ' ' + sub_words[i])
-
-                # Split long lines, and each newline found in the current word
-                if current_width > line_width or i > 0:
-                    wrapped_str += line + '\n'
-                    line = ""
-
-                if line == "":
-                    line = sub_words[i]
-                else:
-                    line += ' ' + sub_words[i]
 
-        return wrapped_str + line
+        # New lines in the user's text force a split
+        unwrapped_lines = self.get_text().split('\n')
+
+        # Now wrap each individual unwrapped line
+        for unwrapped_line in unwrapped_lines:
+
+            sub_words = unwrapped_line.split(' ')
+            # Remove items from sub_words as we go, so stop when empty
+            while len(sub_words) > 0:
+                if len(sub_words) == 1:
+                    # Only one word, so just add it to the end
+                    wrapped_str += sub_words.pop(0)
```

Comment:
What about having wrapped_lines as a list of strings that you "\n".join() at the very end (or even just return the list of lines and change the call site?  I think right now one difficulty to follow the flow of the algorithm is that the place where the linebreaks are introduced is a bit hidden (it's a `+ '\n'` that hides at the end of a line).